hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -287,13 +290,19 @@ public final void register(Class type, Serializer serializer) {
     }
 
     public final void registerGlobal(final Serializer serializer) {
+        registerGlobal(serializer, false);
+    }
+
+    public final void registerGlobal(final Serializer serializer, boolean overrideJavaSerialization) {
```

Comment:
Imho we should pull out the serialization resolution into a separate data-structure to make the SerializationService a bit cleaner. But not part of this PR.